CustomDimensions.php

```diff
@@ -280,10 +281,13 @@ public function addConversionInformation(&$conversion, $visitInformation, Tracke
 
         foreach ($dimensions as $dimension) {
             $index = (int) $dimension['index'];
-            if ($dimension['scope'] === self::SCOPE_VISIT && in_array($index, $conversionIndexes)) {
+            if (in_array($dimension['scope'], array(self::SCOPE_ACTION, self::SCOPE_VISIT)) && in_array($index, $conversionIndexes)) {
                 $field = LogTable::buildCustomDimensionColumnName($dimension);
 
-                if (array_key_exists($field, $visitInformation)) {
+                if ($dimension['scope'] === self::SCOPE_ACTION && array_key_exists($field, $actionDimensions)) {
```

Comment:
I just realize to implement this properly we would actually need to update Piwik core and post the `$action` as fourth parameter in the event `Tracker.newConversionInformation` in `GoalManager.php`.  Then we could here try to get any custom fields like `$action->getCustomFields()`.

While the suggestion would certainly work (haven't tested it though) there might be some edge cases possible where it doesn't. For example after the Custom Dimension Request Processor is executed, other plugins could change the set dimensions via `$action->setCustomField`. This means we need to get the custom dimensions from the action to make sure we get the right value. Didn't notice that before. I'll make a PR for this to post the action as fourth parameter and then we can get the values from there and then we also don't need to make the other method public.